python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py

```diff
@@ -36,6 +36,9 @@ def __init__(
             model_id {str} -- Hugging Face model card string, see
                 https://huggingface.co/models
             device {Optional[int]} -- Device to run the model on, -1 for CPU, 0+ for GPU.
+                                   -- None if using device_map instead. (If both device and device_map
+                                      are specified, device overrides device_map. If unintended, 
```

Comment:
@dmytrostruk, Here's a reference from HF pipeline source [Link](https://github.com/huggingface/transformers/blob/v4.34.0/src/transformers/pipelines/__init__.py#L527): 

```
if device_map is not None:
        if "device_map" in model_kwargs:
            raise ValueError(
                'You cannot use both `pipeline(... device_map=..., model_kwargs={"device_map":...})` as those'
                " arguments might conflict, use only one.)"
            )
        if device is not None:
            logger.warning(
                "Both `device` and `device_map` are specified. `device` will override `device_map`. You"
                " will most likely encounter unexpected behavior. Please remove `device` and keep `device_map`."
            )
        model_kwargs["device_map"] = device_map
```
> 3. `sk_hf.HuggingFaceTextCompletion('model-id', pipeline_kwargs=dict(device_map='auto'))` - since `device` parameter in constructor has default value `-1`, both `device` and `device_map` parameters will be used. Is this expected?

HF overrides device_map if device is not None after throwing a warning.
In our code, device will default to using CPU, meaning specifying the device_map param in pipeline would lead to HF conflict. Ideally we would default device to None rather than -1, but that would mean, we are expecting the user to manually type in "cpu". My fix here expects the user to manually type in `device=None` if device_map is being used so that the conflict doesn't take place.

On second thought, I think it may be better to raise an invalid configuration exception to prompt the user into making a choice rather than going forward with the additional "If". Let me know if that's okay!.

Reply:
@AdarshAcharya5 Thanks for explanation!

> Ideally we would default device to None rather than -1, but that would mean, we are expecting the user to manually type in "cpu".

I think that default device to None looks better than -1. The logic could be like following:
- If `device` is not provided, `device_map` is provided - use `device_map`.
- If `device` is provided, `device_map` is not provided - use `device`.
- If `device` is not provided, `device_map` is not provided - set `device` to `cpu` by default and use `device`.
- If `device` is provided, `device_map` is provided - use both of them. If HF source allows this way of configuration with warning, it means that we should follow the same pattern. Otherwise, the exception should be thrown on HF side. We can leave parameter comment that usage of both parameters at the same time is not recommended.

Please let me know if that makes sense. Thank you! 

Reply:
@dmytrostruk. Sure, I'll make the changes soon!.

Reply:
@dmytrostruk I have made the changes. Please let me know if they're fine!. Also, sorry for the delay, I didn't have access to my machine for a while.

Reply:
@dmytrostruk I have made the changes, please let me know if they're fine!. Apologies for delay, I didn't have access to my machine for a while.

Reply:
@AdarshAcharya5 No worries at all, thanks for your contribution! I pushed small change to fix code formatting. But in general it looks good to me!